app/code/Magento/Sitemap/Controller/Adminhtml/Sitemap/Generate.php

```diff
@@ -23,7 +44,14 @@ public function execute()
         // if sitemap record exists
         if ($sitemap->getId()) {
             try {
+                //We need to emulate to get the correct frontend URL for the product images
+                $this->appEmulation->startEnvironmentEmulation(
+                    $sitemap->getStoreId(),
+                    \Magento\Framework\App\Area::AREA_FRONTEND,
+                    true
+                );
                 $sitemap->generateXml();
+                $this->appEmulation->stopEnvironmentEmulation();
```

Comment:
As mentioned by @JeroenVanLeusden I think it would be a good idea to make sure we are stopping this emulation in all cases including exception cases. https://github.com/magento/magento2/pull/15689#issuecomment-394291002